webui/react/src/components/WorkspaceCreateModal.tsx

```diff
@@ -86,23 +74,21 @@ const WorkspaceCreateModalComponent: React.FC<Props> = ({ onClose, workspaceId }
     [form],
   );
 
+  const workspace = useObservable(workspaceStore.getWorkspace(workspaceId));
   useEffect(() => {
-    initFields(workspace);
-  }, [workspace, initFields]);
-
-  useEffect(() => {
-    fetchWorkspace();
-  }, [workspaceId, fetchWorkspace]);
+    initFields(Loadable.getOrElse(undefined, workspace));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
**question:** is this because `workspace` changes references (even when `workspaceId` doesn't)?

Reply:
yes the workspace observable triggers this often, maybe when it's updated from the API?

I decided to use our `loadableWorkspace = useObservable(...); workspace = Loadable.getOrElse(undefined, loadableWorkspace)` format then I can safely use the workspace variable as a dependency

Reply:
Since we're polling for workspace now we should avoid actually updating the data if nothing has changed.

Reply:
I could go back to `workspaceId` as the dependency, I guess the issue would be if another user changes a workspace name and we update the workspace store, then we want to change the initial values of this form

Reply:
We should fix the polling instead of having potentially incorrect behavior here.

Reply:
I think this is fulfilled in this new commit; it can't be precisely the same as the users store as we currently use an array and not a Map https://github.com/determined-ai/determined/pull/6880/commits/12eeefa9ecbce6ef2120dff59caac240ec84008b

Reply:
Not sure if the workspace order is stable but I'll assume it is for the purposes of my review.